google-cloud-firestore/src/main/java/com/google/cloud/firestore/Query.java

```diff
@@ -1021,11 +1064,33 @@ public Query endAt(@Nonnull DocumentSnapshot snapshot) {
     }
 
     if (options.getStartCursor() != null) {
-      structuredQuery.setStartAt(options.getStartCursor());
+      if (LimitType.Last.equals(options.getLimitType())) {
+        // Swap the cursors to match the flipped query ordering.
+        Cursor cursor =
+            options
+                .getStartCursor()
+                .toBuilder()
+                .setBefore(!options.getStartCursor().getBefore())
+                .build();
+        structuredQuery.setEndAt(cursor);
+      } else {
+        structuredQuery.setStartAt(options.getStartCursor());
+      }
```

Comment:
nit: change to use switch statement on `options.getLimitType` since `LimitType` is an enum

Reply:
Done